packages/roosterjs-content-model-dom/lib/formatHandlers/block/directionFormatHandler.ts

```diff
@@ -14,7 +14,7 @@ export const directionFormatHandler: FormatHandler<DirectionFormat> = {
     },
     apply: (format, element) => {
         if (format.direction) {
-            element.style.direction = format.direction;
+            element.dir = format.direction;
```

Comment:
`dir` supports `rtl | ltr | auto`. `auto` is converted to `ltr` in this format handler

Reply:
I feel "auto" should be mostly used by building a web page, but not editing content.

For editor, the style should be predictable, which means editor should have full control of its style, but not leverage browser's magic behavior. When set to "auto", will the direction automatically changed while user is typing in some RTL language? Say I just type some English, then switch to type Arabic, then suddenly the layout is changed since browser thinks there are more Arabic than English?

So I'll say, in roosterjs, we don't support "dir=auto".

Reply:
Sorry I missed the message.

We want to support auto because on android, due to the custom keyboards, it's hard to know what language the users are using, so we have to depend on the browser to decide the text direction. 

We supported different text direction by paragraphs. To implement it, we set `auto` on new paragrahs.

About more detail about android side, @Rain-Zheng could you help provide more information?

Reply:
Thanks @haven2world for helping explain this! Since Android IME doesn't provide any API for getting the input language, we have to use `auto` as a workaround. By setting this, Chrome will automatically adjust the layout of a new paragraph according to the language. If the paragraph already contains some content and users switch to another language with a different direction, the layout won't be changed. 

I tested applying the `dir="auto"` to editor container div, unfortunately it doesn't work for Android: 

|HTML|Result|
|--|--|
|<img width="573" alt="image" src="https://github.com/microsoft/roosterjs/assets/67583056/2899798f-326a-4ddf-bb4d-944a2b8010ae">|<img width="505" alt="image" src="https://github.com/microsoft/roosterjs/assets/67583056/ab8aa0bc-204c-42d5-93a8-9fddd7efa695">|

Reply:
Can you try to test each minimum repro in what case it work and what case it doesn't?

For example
```html
<div dir="auto">[RTL text here]</div>

<div dir="auto"><span>[RTL text here]</span></div>

<div dir="auto"><div>[RTL text here]</div></div>

<div dir="auto"><table><tr><td>[RTL text here]</td></tr></table></div>

<div dir="auto"><blockquote>[RTL text here]</blockquote></div>

<div dir="auto"><ol><li>[RTL text here]</li></ol></div>

<div dir="auto">[RTL text here]<div>[RTL text here]</div>[RTL text here</div></div>

<div dir="auto">[RTL text here]<br>regular text here</div>

<div dir="auto">regular text here<br>[RTL text here]</div>
```

I want to get deeper understand about this property

Reply:
On the other hand, you can always override format handler using EditorOptions to pass in your customized format handler. That allows you create you own rendering/parsing logic

Reply:
I created a fiddle here: https://jsfiddle.net/pjnq1o9c/

According to the results, these cases are workinig as expected: 

```
<div dir="auto">[RTL text here]</div>

<div dir="auto"><span>[RTL text here]</span></div>

<div dir="auto"><div>[RTL text here]</div></div>

<div dir="auto"><table><tr><td>[RTL text here]</td></tr></table></div>

<div dir="auto"><blockquote>[RTL text here]</blockquote></div>

<div dir="auto"><ol><li>[RTL text here]</li></ol></div>

<div dir="auto">[RTL text here]<div>[RTL text here]</div>[RTL text here]</div>
```
And the following cases are problematic:

```
<div dir="auto">[RTL text here]<br>regular text here</div>

<div dir="auto">regular text here<br>[RTL text here]</div>

<div dir="auto"><div>regular text here</div><div>[RTL text here]</div></div>
```

Currently we are overriding the rooster's default direction handler as a workaround.

Reply:
Ok. I think the principal here is, browser may count the all character under "dir=auto", then decide the direction for this whole container, but not for each child element.

Since the overriding works, I'll treat this as solved. RoosterJs only provide support for the most common scenarios. For each specific feature, if the common support is not enough, you need to create your override or extension. In OWA we also have a lot of override that only needed by OWA, but not common roosterjs. The most we can do is to provide enough entry point to override, but not take every specific requirement into roosterjs.